### PR TITLE
Add starknet_traceBlockTransactions 

### PR DESCRIPTION
**Pull request by mikiw**
## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] No unused dependencies - `./scripts/check_unused_deps.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [x] Linked the issues which this PR resolves
- [x] Checked the TODO section in README.md if this PR resolves it
- [x] Updated the tests
- [x] All tests are passing - `cargo test`

### CODE REVIEW

File: crates/starknet-server/tests/test_trace.rs [hunk 1]
```diff
@@ -189,4 +188,54 @@ mod trace_tests {
...
+        // this test also needs to be updated
+        let block_traces = &devnet
+            .send_custom_rpc("starknet_traceBlockTransactions", json!({ "block_id": "latest" }))
+            .await["result"];
+
+        // assert transaction hash
+        assert_eq!(
+            mint_tx_hash,
+            FieldElement::from_hex_be(
+                block_traces["traces"][0]["transaction_hash"].as_str().unwrap()
```
Comment (FabijanC) [inline crates/starknet-server/tests/test_trace.rs:208]: Is len of `block_traces["traces"]` supposed to be one? If so, I would add a length assertion before all the assertions, and extract `block_traces["traces"][0]` into a `trace` variable to reduce duplication.

**Review by FabijanC — COMMENTED** [review]
Looks good, just one question (potential change request) in the tests.

File: crates/starknet-server/tests/test_trace.rs (hunk 1 above)
Comment (mikiw) [reply crates/starknet-server/tests/test_trace.rs]: addressed in cdfa97f922e9acc47385258a71ad42a3195bfae9

Comment (FabijanC) [reply crates/starknet-server/tests/test_trace.rs]: If you already deserialize it with
```
serde_json::from_value::<Vec<BlockTransactionTrace>>(block_traces.clone(),)
```
why not later use `validate_invocation`, `execute_invocation` and `fee_transfer_invocation` of that deserialized object that is only used for length assertion?

Also, no need to call the variable `traces` if it's just one `trace` (the one at index 0)

Comment (mikiw) [reply crates/starknet-server/tests/test_trace.rs]: that would require if let type match which would increase code size

Comment (FabijanC) [reply crates/starknet-server/tests/test_trace.rs]: I see. but why can't the length be asserted on the object whose property [0] you are taking? I.e. without deserialization?

Comment (mikiw) [reply crates/starknet-server/tests/test_trace.rs]: because it's `&Value` type

Comment (FabijanC) [reply crates/starknet-server/tests/test_trace.rs]: ok then

**Review by FabijanC — APPROVED** [review]

**Pull request merged**